replica/table.cc

```diff
@@ -1379,10 +1379,12 @@ future<std::unordered_set<sstring>> table::take_snapshot(database& db, sstring j
     co_await io_check(sync_directory, jsondir);
 
     std::unordered_set<sstring> table_names;
+    table_names.reserve(tables.size());
     for (auto& sst : tables) {
         auto f = sst->get_filename();
         auto rf = f.substr(sst->get_dir().size() + 1);
         table_names.insert(std::move(rf));
+        co_await coroutine::maybe_yield();
```

Comment:
if we consider each iteration takes 1us, then we indeed have a potential for few ms stalls here.

Reply:
thousand sstable threshold can be easily hit with off-strategy compaction.

Reply:
right

Reply:
I don't think an iteration can take a microsecond here, but okay.